src/common/com/intellij/plugins/haxe/ide/module/HaxeModuleBuilder.java

```diff
@@ -51,6 +74,75 @@ public void moduleCreated(@NotNull Module module) {
     final CompilerModuleExtension model = (CompilerModuleExtension)CompilerModuleExtension.getInstance(module).getModifiableModel(true);
     model.setCompilerOutputPath(model.getCompilerOutputUrl());
     model.inheritCompilerOutputPath(false);
+
+    final Project project = module.getProject();
+    List<Pair<String, String>> sourcePaths = getSourcePaths();
+    String srcPath = null;
+
+    if ((sourcePaths.size() > 0)) {
+      srcPath = sourcePaths.get(0).getFirst();
+      VirtualFile dir = VirtualFileManager.getInstance().findFileByUrl(VfsUtilCore.pathToUrl(srcPath));
+
+      if (dir != null) {
+        try {
+          VirtualFile mainClassFile = dir.findFileByRelativePath("Main.hx");
+          if (mainClassFile == null) {
+            HaxeFileTemplateUtil
+              .createClass("Main", "", PsiManager.getInstance(project).findDirectory(dir), "HaxeMainClass",
+                           null);
+          }
+        }
+        catch (Exception e) {
+          e.printStackTrace();
+        }
+      }
+
+      HaxeModuleSettings moduleSettings = HaxeModuleSettings.getInstance(module);
+      moduleSettings.setMainClass("Main");
+      moduleSettings.setArguments("");
+      moduleSettings.setNmeFlags("");
+      moduleSettings.setHaxeTarget(HaxeTarget.NEKO);
```

Comment:
Out project(s) have something like 80 modules, depending upon the target.  Basically, we think of modules as libraries.  We have them compile into their own haxelibs, but they are part of the larger project  so that we can traverse the sources easily.  Each one of them should not have a main class.

From: as3boyan [mailto:notifications@github.com]
Sent: Tuesday, February 02, 2016 10:46 AM
To: TiVo/intellij-haxe
Cc: Eric Bishton
Subject: Re: [intellij-haxe] CreateMainClassAndOpenItInEditor (#398)


In src/common/com/intellij/plugins/haxe/ide/module/HaxeModuleBuilder.java<https://github.com/TiVo/intellij-haxe/pull/398#discussion_r51614372>:

> +          if (mainClassFile == null) {

> +            HaxeFileTemplateUtil

> +              .createClass("Main", "", PsiManager.getInstance(project).findDirectory(dir), "HaxeMainClass",

> +                           null);

> +          }

> +        }

> +        catch (Exception e) {

> +          e.printStackTrace();

> +        }

> +      }

> +

> +      HaxeModuleSettings moduleSettings = HaxeModuleSettings.getInstance(module);

> +      moduleSettings.setMainClass("Main");

> +      moduleSettings.setArguments("");

> +      moduleSettings.setNmeFlags("");

> +      moduleSettings.setHaxeTarget(HaxeTarget.NEKO);

That is interesting, I just don't use multiple modules, so I don't know how it is in practical sense

—
Reply to this email directly or view it on GitHub<https://github.com/TiVo/intellij-haxe/pull/398/files#r51614372>.

________________________________

This email and any attachments may contain confidential and privileged material for the sole use of the intended recipient. Any review, copying, or distribution of this email (or any attachments) by others is prohibited. If you are not the intended recipient, please contact the sender immediately and permanently delete this email and any attachments. No employee or agent of TiVo Inc. is authorized to conclude any binding agreement on behalf of TiVo Inc. by email. Binding agreements with TiVo Inc. may only be made by a signed written agreement.